lib/push.py

```diff
@@ -116,12 +116,8 @@ def prepare(self):
       ch.INFO("preparing metadata")
       self.image.metadata_load()
       hist = self.image.metadata["history"]
-      for i in range(len(hist)):
+      for i in range(len(hist) - 1):
```

Comment:
If the last history entry happens to have the `empty_layer` present then we can hit a corner case where will have a layerless image. IIRC the nginx image history has a final entry with an empty layer. So if we built an image with something silly like:

```
FROM nginix
```

We may bump into this issue. That is why I popped the last entry instead of iterating to n -1. Perhaps there is a better way to handle this?